azurelinuxagent/ga/exthandlers.py

```diff
@@ -1259,7 +1260,7 @@ def download(self):
                 self.logger.info("The existing extension package is invalid, will ignore it.")
 
         if not package_exists:
-            self.protocol.client.download_extension(self.pkg.uris, destination, on_downloaded=lambda: self._unzip_extension_package(destination, self.get_base_dir()))
+            self.protocol.get_goal_state().download_extension(self.pkg.uris, destination, on_downloaded=lambda: self._unzip_extension_package(destination, self.get_base_dir()))
```

Comment:
We need to get rid of WireProtocol.get_goal_state(), since it returns a global. That requires deeper refactoring, using it for the moment